### PR TITLE
HPCC-18512 Switch to WebPack for ECL Watch build

### PR DESCRIPTION
**Pull request by GordonSmith**
* Start using NPM for third party packages
* Call "npm install" as part of the build process
* Refactored code so WebPack can spot dynamic loading
* Refactored resources to simplify deploy
* Added missing dojox css directly into hpcc.css

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq Please review and test the new build process.

**Comment by richardkchapman** [comment]
Can you kill the - in the commit message after the Jira number

**Comment by GordonSmith** [comment]
Done

File: esp/src/eclwatch/DFUSearchWidget.js [hunk 1]
```diff
@@ -69,57 +51,6 @@ define([
         postCreate: function (args) {
             this.inherited(arguments);
             this.borderContainer = registry.byId(this.id + "BorderContainer");
-            //this.tabContainer = registry.byId(this.id + "TabContainer");
```
Comment (miguelvazq) [inline esp/src/eclwatch/DFUSearchWidget.js]: Remind me, where is all this getting moved to?

**Review by miguelvazq — COMMENTED** [review]
See my question.

File: esp/src/eclwatch/DFUSearchWidget.js (hunk 1 above)
Comment (GordonSmith) [reply esp/src/eclwatch/DFUSearchWidget.js]: In this case I am simply deleting old and unused commented code.

None of the ECL Watch functionality is moving to @hpcc-js in this pull request.

**Comment by miguelvazq** [comment]
Code looks good. Let me build and test.

**Comment by miguelvazq** [comment]
Built, and tested. Some sources were missing, did a second build and sources resolved and ECLWatch working as expected. Perhaps a network issue caused the build.sh to not perform a step?

**Comment by GordonSmith** [comment]
@miguelvazq FYI I tweaked how the build is being called (moved it out of build.sh and into CMakeList.txt).  I also fixed some linting issues and updated travis.yml so that travis will auto test , lint and build again.

**Comment by GordonSmith** [comment]
@miguelvazq bump

**Comment by miguelvazq** [comment]
Installed on fresh environment with no previous files/directories worked as expected.

**Comment by miguelvazq** [comment]
Looks good.

**Pull request merged**